### DISCUSSION TITLE
Touch Pins on Audiokit

### DISCUSSION THREAD

**Discussion by mydroidbox** [pschatzmann/arduino-audio-tools#628] [Q&A] [answered]
Greetings Phil.
Is there any way to get any of the touch pins to work while the SD card is
inserted. I can only seem to get pins 12 and 13 working for touch with SD card ejected. 
Pin 12 will work with the usual sd card dip positions of 01100 in a touch sketch with SD card inserted but once used in a sketch where the sd card is accessed the touch reading drops to zero.

**Comment by pschatzmann** [comment, marked as answer]
All boards are reusing the same pins for different functions and it depends pretty much on your model (and used sd API) which conflicts you have!
You can double check this in the pin definitions (and conflicts) of the board_def.h of your model.

So in the end you need to choose which function you want to use (at the cost of the other)

**Reply by mydroidbox** [reply]
The SPI pinout is easy enough to change on a regular esp32 dev board , is it fixed on these boards or could I move CS to a different pin for example and try to figure out which dip switch completes the path. I'm using a version 2.2 with board setting 5.

**Comment by pschatzmann** [comment]
I doubt it, but why don't you give it just a try...

**Reply by mydroidbox** [reply]
Ok I've moved cs to IO4 (soldered) and edited the board manager definition to 
#define PIN_AUDIO_KIT_SD_CARD_CS 4. 
I'm getting the error message:  AudioSourceSD.h : 49 - SD.begin cs=13 failed.
 However I can't find where else cs is defined as 13.  let me know if you know where
I might find this definition. Thanks

**Comment by pschatzmann** [comment]
As mentioned above it is in the [board_def](https://github.com/pschatzmann/arduino-audiokit/blob/main/src/audio_board/ai_thinker_es8388_3478/board_def.h). For 5 it is in ai_thinker_es8388_3478.

But why don't you just the SD example from the ESP32 standard examples by adding the SD initialization as described [here](https://www.pschatzmann.ch/home/2021/12/06/the-ai-thinker-audio-kit-experience-or-nothing-is-right/)

**Reply by mydroidbox** [reply]
sorry I mistyped above where i wrote board manager I actually meant board definition.
What you wote above is exactly what i was saying i did in my last post but I had edited the wrong file. I had edited the ai_thinker_es8388_2957 board file so thanks for clearing that up and it worked right away. So I now have touch working on PIn 13 and still able to access the sd card using SPI.begin(14, 2, 15, 4);  Dip switch 2 (on my board at least) connects SD cs pin to IO13 so that should be turned off.
